meteor/client/ui/FloatingInspectors/NoraFloatingInspector.tsx

```diff
@@ -108,20 +155,56 @@ export class NoraPreviewRenderer extends React.Component<{}, IStateHeader> {
 				this.postNoraEvent(e.contentWindow, noraContent)
 			}
 		}, 1000)
+
+		// set up IntersectionObserver to keep the preview inside the viewport
+		let options = {
+			threshold: [] as number[],
+		}
+		for (let i = 0; i < 50; i++) {
+			options.threshold.push(i / 50)
+		}
+
+		this._observer = new IntersectionObserver((entries) => {
+			entries.forEach((entry) => {
+				if (entry.target === e) {
+					this._intersections = {
+						boundingClientRect: entry.boundingClientRect,
+						intersectionRect: entry.intersectionRect,
+					}
+				}
+			})
+		}, options)
+		this._observer.observe(e)
 	}
 
-	render() {
-		if (!this.state) return null
+	componentWillUnmount() {
+		this._observer.disconnect()
+	}
 
+	getElStyle() {
 		const style = { ...this.state.style }
 		style.visibility = this.state.show ? 'visible' : 'hidden'
 
+		if (this.state.flip.x && this.state.flip.y) {
+			style.transform = 'translate(0, 2em)'
```

Comment:
I think it would be nice to move out this `2em` constant value to a CSS variable in `.segment-timeline__mini-inspector--graphics--preview` in `rundownView.scss` perhaps?